app/models/sushi.rb

```diff
@@ -61,4 +61,79 @@ def coerce_data_types(params = {})
       @data_types = Array.wrap(params[:data_type]&.split('|')).map(&:downcase)
     end
   end
+
+  # this module accounts for data manipulation that needs to happen across all reports.
+  module AttributePerformance
+    extend ActiveSupport::Concern
+    included do
+      attr_reader :begin_date
+      attr_reader :end_date
+    end
+
+    def attribute_performance_for_resource_types
+      data_for_resource_types.group_by(&:resource_type).map do |resource_type, records|
+        { "Data_Type" => resource_type || "",
+          "Access_Method" => "Regular",
+          "Performance" => {
+            "Total_Item_Investigations" =>
+            records.each_with_object({}) do |record, hash|
+              hash[record.year_month.strftime("%Y-%m")] = record.total_item_investigations
+              hash
+            end,
+            "Total_Item_Requests" =>
+            records.each_with_object({}) do |record, hash|
+              hash[record.year_month.strftime("%Y-%m")] = record.total_item_requests
+              hash
+            end
+          } }
+      end
+    end
+
+    def attribute_performance_for(resource_type:)
```

Comment:
updated to accept a param so its reusable

Reply:
I think this might introduce conditions for further introduction of a very subtle bug.  In adding the parameter, we're adjusting the label but we're not adjusting the SQL we're querying.  

The internal data for this method is very specifically a query for the platform.  With the method signature, it looks like I could give it "book" and expect to only have "book" results, even though we'd be querying the platform.

Reply:
taking a closer look at this made me realize I left some things out. marked back as a draft!